### PR TITLE
Add in the test_energy.

### PR DESCRIPTION
**Pull request by dingzw007**
I added in the test_energy.py file to test the  energy. It pass on my end. This should increase the code convergence to a higher portion rates.

### CODE REVIEW

**Comment by kbsezginel** [comment]
This looks great but I think it needs cleaning up.

**Comment by kbsezginel** [comment]
Also, seems like tests are failing

File: .travis.yml [hunk 1]
```diff
@@ -33,7 +33,7 @@ before_script:
 
 script:
   - pip install -e .
-  - py.test -v --cov=./
```
Comment (kbsezginel) [inline .travis.yml]: Why did you change this line?

File: output [hunk 2]
```diff
@@ -0,0 +1,47 @@
+    Molecular point group: c2v
```
Comment (kbsezginel) [inline output]: I don't think we need this file in the repository.

File: qm7/scf.py [hunk 3]
```diff
@@ -21,6 +21,7 @@ def __init__(self, cli_args):
         self.damp_start = cli_args.dampstart
         self.damping = cli_args.damping
         self.DIIS = cli_args.DIIS
+
```
Comment (kbsezginel) [inline qm7/scf.py]: There should be only one empty new line between class functions.

File: tests/test_energy.py [hunk 4]
```diff
@@ -0,0 +1,44 @@
+"""
+this is to test the eneryg of H2O with sto-3g basis
+
+"""
+
+from qm7.scf import SCF
+import psi4
+import argparse
+
```
Comment (kbsezginel) [inline tests/test_energy.py]: There should be two new lines here (PEP8).

File: .travis.yml (hunk 1 above)
Comment (dingzw007) [reply .travis.yml]: I only remove the "-v" from that file, as it will confuse with our command line. It should affect the testing as it only change the content to show. Here "-v" means verbose. ..

File: tests/test_energy.py [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+    parser.add_argument('--damping', action='store_true', default=False,
+                    help='Use damping (default: False)')
+    parser.add_argument('--dampvalue', '-dv', type=float, default=0.2, metavar='',
+                    help='Damping value (default: 0.2)')
+    parser.add_argument('--dampstart', '-ds', type=int, default=5, metavar='',
+                    help='Iteration to start damping (default: 5)')
+    parser.add_argument('--DIIS', '-D', action='store_true', default=False,
+                    help='Iteration to start damping (default: 5)')
+    
+    parser.add_argument('--verbose', '-v', type=str, action=None, default="false")  # For tests
```
Comment (kbsezginel) [inline tests/test_energy.py]: We don't need any of the parser code here. You can just give the arguments you want and run the scf as:
```python
scf = SCF(args)
scf.initialize()
scf.solve()
```

Take a look at here: [How to write tests for argparse](https://stackoverflow.com/questions/18160078/how-do-you-write-tests-for-the-argparse-portion-of-a-python-module)

File: tests/test_method.py [hunk 6]
```diff
@@ -0,0 +1 @@
+pass
```
Comment (kbsezginel) [inline tests/test_method.py]: We don't need this empty test file. Just delete it.

File: .travis.yml (hunk 1 above)
Comment (dingzw007) [reply .travis.yml]: Looks like it is the "--conv" caused the trouble....

File: tests/test_method.py (hunk 6 above)
Comment (dingzw007) [reply tests/test_method.py]: Sure~

**Comment by kbsezginel** [comment]
Actually now that I think about it I think it's a good idea to use pytest parametrize (see below) with different options. We can do things like testing with/without damping, DIIS. We can also try different basis sets as well. Also, we might add another test molecule.

These don't need to be in this pull request but once you clean it up we can do those. I am gonna create issues for those.

[Parametrizing fixtures and test functions](https://docs.pytest.org/en/latest/parametrize.html)

File: tests/test_energy.py (hunk 5 above)
Comment (dingzw007) [reply tests/test_energy.py]: Thanks~~~

File: .travis.yml (hunk 1 above)
Comment (kbsezginel) [reply .travis.yml]: If -v is just for verbose then I guess it should not matter.

File: tests/test_energy.py (hunk 5 above)
Comment (kbsezginel) [reply tests/test_energy.py]: I guess a quick and dirty way to do this might be something like:

```python
args = argparse.Namespace()
args.molecule = 'tests/water'
args.basis = 'sto-3g'
args.nocc = 5
args.max_iter = 50
args.e_conv = 1e-6
args.d_conv = 1e-6
args.dampvalue = 0.2
args.dampstart = 5
args.damping = False
args.DIIS = True

scf = SCF(args)
scf.initialize()
scf.solve()

psi4.set_options({"scf_type":"pk"})
psi4_energy = psi4.energy("SCF/sto-3g", molecule=scf.mol)
assert np.allclose(scf.Etotal, psi4_energy)
```

The part where we define the arguments can be a separate function in the future.

**Comment by kbsezginel** [comment]
Looks good to me!
